src/Chainweb/Mempool/InMem.hs

```diff
@@ -243,7 +243,7 @@ markValidatedInMem logger tcfg lock txs = withMVarMasked lock $ \mdata -> do
     logg Info $ "mark " <> sshow (length (V.zip expiries hashes)) <> " txs as validated"
     x <- readIORef curTxIdxRef
     logg Info $ "previous current tx index size: " <> sshow (currentTxsSize x)
-    x' <- flip currentTxsInsertBatch (V.zip expiries hashes) x
+    x' <- currentTxsInsertBatch x (V.zip expiries hashes)
```

Comment:
Should this perhaps have a bang pattern? Or maybe below: `writeIORef curTxIdxRef $! x'`

Reply:
Good catch!